src/main/java/relay/data_access/FirebaseCourseDataAccessObject.java

```diff
@@ -26,14 +26,19 @@ public FirebaseCourseDataAccessObject() {
      * @param course The Course object containing course details to be added to Firestore.
      * @throws RuntimeException If an InterruptedException or ExecutionException occurs during Firestore operations.
      */
-    public void createCourse(Course course) throws InterruptedException, ExecutionException {
+    public void saveCourse(Course course) {
```

Comment:
Rename to `save`.

Reply:
Also, you need to check if it the provided course exists first, if it doesn't then do the `db.collection('courses').add` otherwise use the `update method`, see firebase docs